avm/utilities/e2e-template-assets/templates/azure-stack-hci/modules/azureStackHCIClusterPreqs/ashciPrereqs.bicep

```diff
@@ -0,0 +1,309 @@
+param location string
```

Comment:
If these files are only used by the azure-stack-hci module, they're should be defined there, not in the general templates. Each change to the files in the `e2e-teplate-assets` folder triggers new test deployments for **all** modules

Reply:
Also, I'd highly recommend reducing the amount of parameters to the minimum absolutely necessary (usually only the top-level resource names) to deploy the dependencies, and for those add also decriptions.

Reply:
Mind you, this comment applies to all test files (main.test.bicep / main.test.bicep / assets)

Reply:
@AlexanderSehr These templates are used for all HCI-related e2e testing (everything needs a cluster to test on)--Arc VM instance, NICs, Disks, Logical Networks, AKS clusters, AVD, etc.... Given the complexity, I'm hoping to only need to maintain one copy--is there another approach/location you can recommend?

Reply:
Hey @mbrat2005, I tend to agree with @AlexanderSehr here, especially as keeping the files within the module's boundaries would allow the module owner to fully maintain their lifecycle. 
However it's also a pity to duplicate the same code for multiple tests.
Between the two alternatives I'd currently vote for the former (keeping them in the module even at the cost of duplication).

I'm not sure if having a template-asset subfolder in the e2e tests folder and reference files there is currently supported (I'd need to double check the CI code and test it myself). If so, I would not have anything against this approach. Maybe something to investigate.

Reply:
I'll test moving the shared templates to a subdirectory of the e2e module directory--if I am reading the CI code right, it only looks for directories with main.test.bicep, so this could work. 

That said, I'm certain that what will require maintenance on these HCI related modules is this set of shared templates, rather than the BRM modules themselves. When this breaks because the PG has introduced some new bug or feature, we'll need PRs for each module and have the ongoing challenge of keeping these in sync. 

What are your thoughts on hosting these templates in a separate github repo and referencing with a Microsoft.Resources/deployments resource (if that is possible...)? @AlexanderSehr @eriqua

Reply:
Glat it works @mbrat2005, thanks for addressing and investigating! 💪 

@AlexanderSehr I think we should define a naming convention for the shared assets folder at module level, so that this can be followed consistently where the same scenario applies.

To the proposal of using a separated repo I'd probably vouch against that as it would create an external dependency and not help with maintenance.